src/beliefs/discrete.jl

```diff
@@ -36,6 +36,15 @@ function DiscreteBelief(pomdp, b::Vector{Float64}; check::Bool=true)
     return DiscreteBelief(pomdp, ordered_states(pomdp), b)
 end
 
+function DiscreteBelief(pomdp, b; check::Bool=true)
```

Comment:
We should avoid calling ordered_states twice since it can be expensive. I think that means we need 3 outer constructors (in addition to the automatically created inner constructor)

```julia
DiscreteBelief(pomdp::POMDP, state_list::AbstractVector, b::AbstractVector{Float64}, check::Bool=true) # <-- this is where the checks occur. note here check is NOT a keyword arg.
DiscreteBelief(pomdp::POMDP, b::Vector{Float64}; check::Bool=true) # basically just calls DiscreteBelief(pomdp, ordered_states(pomdp), b, check)
DiscreteBelief(pomdp::POMDP, b; check::Bool=true) # handles any belief
```